### PR TITLE
Upgrade crates with known vulnerabilities

### PR DESCRIPTION
**Pull request by Mateo-mro**
# Description
Upgrade crates with known vulnerabilities

# Changes
Several crates have been updated to avoid known vulnerabilities

## How to test
1. Regression tests passing

### CODE REVIEW

File: Cargo.lock [hunk 1]
```diff
@@ -5215,19 +5230,23 @@ dependencies = [
...
+checksum = "07ecc0cd7cac091bf682ec5efa18b1cff79d617b84181f38b3951dbe135f607f"
 dependencies = [
  "ring",
  "untrusted",
 ]
 
 [[package]]
 name = "whoami"
-version = "1.4.1"
+version = "1.5.0"
```
Comment (Mateo-mro) [inline Cargo.lock:5243]: https://rustsec.org/advisories/RUSTSEC-2024-0020

File: Cargo.lock [hunk 2]
```diff
@@ -5215,19 +5230,23 @@ dependencies = [
 
 [[package]]
 name = "webpki"
-version = "0.22.0"
+version = "0.22.2"
```
Comment (Mateo-mro) [inline Cargo.lock:5233]: https://rustsec.org/advisories/RUSTSEC-2023-0052

File: Cargo.lock [hunk 3]
```diff
@@ -3779,9 +3788,9 @@ dependencies = [
 
 [[package]]
 name = "rustls-webpki"
-version = "0.101.3"
+version = "0.101.4"
```
Comment (Mateo-mro) [inline Cargo.lock:3791]: https://rustsec.org/advisories/RUSTSEC-2023-0053

File: crates/observe/Cargo.toml [hunk 4]
```diff
@@ -6,7 +6,7 @@ edition = "2021"
 license = "MIT OR Apache-2.0"
 
 [dependencies]
-atty = "0.2"
+atty = "0.2.14"
```
Comment (Mateo-mro) [inline crates/observe/Cargo.toml:9]: https://rustsec.org/advisories/RUSTSEC-2021-0145 (still not solved with this update, they are working on it)

File: Cargo.lock [hunk 5]
```diff
@@ -2204,17 +2204,17 @@ checksum = "d2fabcfbdc87f4758337ca535fb41a6d701b65693ce38287d856d1674551ec9b"
 
 [[package]]
 name = "h2"
-version = "0.3.20"
+version = "0.3.25"
```
Comment (Mateo-mro) [inline Cargo.lock:2207]: https://rustsec.org/advisories/RUSTSEC-2024-0003

**Comment by squadgazzz** [comment]
Not sure about upgrading transitive dependencies directly through the `cargo.lock`. How about upgrading libs from the `cargo.toml` that import outdated dependencies?

**Comment by Mateo-mro** [comment]
> Not sure about upgrading transitive dependencies directly through the `cargo.lock`. How about upgrading libs from the `cargo.toml` that import outdated dependencies?

Updating dependencies in `.lock` file when it is minor versions is totally fine and recommendable for the cases in which the `parent` crate didn't release a fix yet (which is very common sadly). I did it like that so I am not doing too many changes, but I am happy to bump all those crates to newest version!

**Review by MartinquaXD — APPROVED** [review]
Nice initiative!
BTW there is also `cargo audit fix` which adds a lot of additional changes to the lockfile. Did you consider using that instead?

**Comment by MartinquaXD** [comment]
It's a separate command that can be added to cargo itself. More information can be found here: https://crates.io/crates/cargo-audit

**Comment by Mateo-mro** [comment]
> It's a separate command that can be added to cargo itself. More information can be found here: https://crates.io/crates/cargo-audit

ah I has to be installed with a feature flag, I see. I didn't know, thanks!! I did it and it didn't update anything else besides what I updated in this PR.

**Review by squadgazzz — APPROVED** [review]

**Pull request merged**